### PR TITLE
Fix warnings in the complete test

### LINKED ISSUE #87: Warnings in complete_test
Several warnings come in the Webots console when launching the `complete_test`.
Most probably the commands sent with ROS have values out of bounds for the motors.

### PR DESCRIPTION
**Pull request by lukicdarkoo**
Fixes #87

### CODE REVIEW

**Comment by lukicdarkoo** [comment]
@BenjaminHug8 There is one warning left but we cannot fix it (missing Joystick)

**Comment by BenjaminHug8** [comment]
I am still blocked in the test with gps speed. I pull and recompile my master branch to test.

**Comment by BenjaminHug8** [comment]
> @BenjaminHug8 There is one warning left but we cannot fix it (missing Joystick)

Yes no problem about this one, I had no choice than to have it when I updated the tests of `complete_test`.

**Comment by lukicdarkoo** [comment]
> I am still blocked in the test with gps speed. I pull and recompile my master branch to test.

I missed to mentioned, there were some bugs in the `ros` controller, you have to checkout to:
https://github.com/cyberbotics/webots/pull/3920

**Comment by BenjaminHug8** [comment]
> I missed to mentioned, there were some bugs in the `ros` controller, you have to checkout to: [cyberbotics/webots#3920](https://github.com/cyberbotics/webots/pull/3920)

Ok I see, I will test that!

**Comment by BenjaminHug8** [comment]
Do you also have `[ERROR] [1637338038.902909262]: Failed to call service robot/wwi_receive_text.` in the ROS terminal ?

File: src/complete_test.cpp [hunk 1]
```diff
@@ -3302,11 +3260,22 @@ int main(int argc, char **argv) {
...
 
   // test restart_controller
+  supervisor_get_from_def_srv.request.name = "ANOTHER_ROBOT";
+  supervisor_get_from_def_client.call(supervisor_get_from_def_srv);
+  uint64_t another_robot_node = 0;
+  if (supervisor_get_from_def_srv.response.node != 0) {
+    ROS_INFO("Got from DEF node: %ld.", supervisor_get_from_def_srv.response.node);
+    another_robot_node = supervisor_get_from_def_srv.response.node;
+  } else
+    ROS_ERROR("Could not get node from DEF.");
```
Comment (BenjaminHug8) [inline src/complete_test.cpp:3270]:
```suggestion
    ROS_ERROR("Could not ANOTHER_ROBOT node from DEF.");
```
Consistency with the other tests.

**Review by BenjaminHug8 — CHANGES_REQUESTED** [review]
Except a minor change that's perfect!

**Comment by BenjaminHug8** [comment]
> Do you also have `[ERROR] [1637338038.902909262]: Failed to call service robot/wwi_receive_text.` in the ROS terminal ?

In case you do so may I ask you also to resolve this?

**Comment by lukicdarkoo** [comment]
> > Do you also have `[ERROR] [1637338038.902909262]: Failed to call service robot/wwi_receive_text.` in the ROS terminal ?
> 
> In case you do so may I ask you also to resolve this?

I cannot reproduce the error locally and I don't see it in [the CI](https://github.com/cyberbotics/webots/runs/4266195903?check_suite_focus=true). I am not sure why are you getting it

**Comment by BenjaminHug8** [comment]
So you do not get this error at the end of the complete test with the `master` branch of Webots?

Screenshot from 2021-11-22 14-00-05

**Review by BenjaminHug8 — APPROVED** [review]
Thank you!

File: src/complete_test.cpp [hunk 2]
```diff
@@ -3302,11 +3260,22 @@ int main(int argc, char **argv) {
...
 
   // test restart_controller
+  supervisor_get_from_def_srv.request.name = "ANOTHER_ROBOT";
+  supervisor_get_from_def_client.call(supervisor_get_from_def_srv);
+  uint64_t another_robot_node = 0;
+  if (supervisor_get_from_def_srv.response.node != 0) {
+    ROS_INFO("Got from DEF node: %ld.", supervisor_get_from_def_srv.response.node);
+    another_robot_node = supervisor_get_from_def_srv.response.node;
+  } else
+    ROS_ERROR("Could not ANOTHER_ROBOT node from DEF.");
```
Comment (omichel) [inline src/complete_test.cpp:3270]: What does this mean? It seems a word is missing here?

File: src/complete_test.cpp [hunk 3]
```diff
@@ -3302,11 +3260,22 @@ int main(int argc, char **argv) {
   time_step_client.call(time_step_srv);
 
   // test restart_controller
+  supervisor_get_from_def_srv.request.name = "ANOTHER_ROBOT";
+  supervisor_get_from_def_client.call(supervisor_get_from_def_srv);
+  uint64_t another_robot_node = 0;
+  if (supervisor_get_from_def_srv.response.node != 0) {
+    ROS_INFO("Got from DEF node: %ld.", supervisor_get_from_def_srv.response.node);
```
Comment (omichel) [inline src/complete_test.cpp:3267]: I would have said "Got node %ld from DEF." instead.

File: src/complete_test.cpp (hunk 2 above)
Comment (BenjaminHug8) [reply src/complete_test.cpp:3270]: It misses a `get` that I forgot in my suggestion on this line.

Comment (omichel) [reply src/complete_test.cpp:3270]: Can we fix it?

File: src/complete_test.cpp (hunk 3 above)
Comment (omichel) [reply src/complete_test.cpp:3267]: As well as this one?

File: src/complete_test.cpp (hunk 2 above)
Comment (omichel) [reply src/complete_test.cpp:3270]: @BenjaminHug8 @lukicdarkoo anyone?

Comment (BenjaminHug8) [reply src/complete_test.cpp:3270]: Should be fixed in #91.

File: src/complete_test.cpp (hunk 3 above)
Comment (BenjaminHug8) [reply src/complete_test.cpp:3267]: Should be fixed in #91.

**Pull request merged**